### ISSUE TITLE
[Feature Request]: Oobabooga API support

### ISSUE THREAD

**Issue by CorentinvdBdO** [run-llama/llama_index#8148]
### Feature Description

Hi! 

Do you pan to add the oobabooga/text-generation-webui inference API into LlamaIndex LLMs ? i use it a lot to test different models and the easiest way for me would be to plug it directly into LlamaIndex.

Thanks for your efforts!

Please forgive me if this is already supported.

**Comment by nerdai** [comment]
Thanks @CorentinvdBdO for submitting this. Would like to get some clarification though.

My (likely naive) understanding here is that `oobabooga/text-generation-webui` supplies a Gradio app that you can run locally to test different LLMs.

Can you elaborate on what you mean here in terms of plugging it directly into this library? 🙏

**Comment by CorentinvdBdO** [comment]
You're right on oobabooga/text-generation-webui, but it also allows to open the model as a blocking or streaming API. Is it possible to handle this API like the OpenAI API? It's an easy way to have a free and model agnostic endpoint. However it may lacks some features like the logits.

**Comment by nerdai** [comment]
Okay then. I wonder if this can just be accomplished with `CustomLLM`? I'll add this to Request For Contribution board in any case.
